R/frosting.R

```diff
@@ -194,6 +194,20 @@ apply_frosting.epi_workflow <- function(workflow, components, the_fit, ...) {
     return(components)
   }
   layers <- workflow$post$actions$frosting$frosting$layers
+
+  # checks if layer_predict() is in the postprocessors
+  list_of_layer_names <- c()
+  for (l in seq_along(layers)){
+    la <- layers[[l]]
+    list_of_layer_names <- append(list_of_layer_names,class(la)[1])
+  }
```

Comment:
While this "works", in R (and other languages), this type of vector-growing-in-a-loop should be avoided at all costs. The reason is that, internally, R has to sequentially hunt for memory at each iteration.

Best practice is:
1. Avoid the loop if possible.
2. If not possible, allocate space for the whole vector before the loop.

To do 2, one would just do `list_of_layer_names <- character(length(layers))`. But 1 is possible. Try replacing these lines with:
```r
layer_names <- map_chr(layers, ~ class(.x)[1])
```

For example:
``` r
A = matrix()
B = data.frame()
C = vector()
D = factor()
l = list(A=A, B=B, C=C, D=D)

map_chr(l, ~ class(.x)[1])
#>            A            B            C            D 
#>     "matrix" "data.frame"    "logical"     "factor"
```

<sup>Created on 2022-06-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>